features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5XPalGattClient.cpp

```diff
@@ -0,0 +1,1616 @@
+    connection_handle_t connection,
+    attribute_handle_t discovery_range_begining
+) {
+    return launch_procedure<DiscoverPrimaryServiceProcedure>(
+        connection, discovery_range_begining
+    );
+}
+
+ble_error_t nRF5XGattClient::discover_primary_service_by_service_uuid(
+    connection_handle_t connection_handle,
+    attribute_handle_t discovery_range_beginning,
+    const UUID& uuid
+) {
+    return launch_procedure<DiscoverPrimaryServiceByUUIDProcedure>(
+        connection_handle, discovery_range_beginning, uuid
+    );
+}
+
+ble_error_t nRF5XGattClient::find_included_service(
+    connection_handle_t connection_handle,
+    attribute_handle_range_t service_range
+) {
+    return launch_procedure<FindIncludedServicesProcedure>(
+        connection_handle, service_range
+    );
+}
+
+ble_error_t nRF5XGattClient::discover_characteristics_of_a_service(
+    connection_handle_t connection_handle,
+    attribute_handle_range_t discovery_range
+) {
+    return launch_procedure<DiscoverCharacteristicsProcedure>(
+        connection_handle, discovery_range
+    );
+}
+
+ble_error_t nRF5XGattClient::discover_characteristics_descriptors(
+    connection_handle_t connection_handle,
+    attribute_handle_range_t descriptors_discovery_range
+) {
+    return launch_procedure<DiscoverDescriptorsProcedure>(
+        connection_handle, descriptors_discovery_range
+    );
+}
+
+ble_error_t nRF5XGattClient::read_attribute_value(
+    connection_handle_t connection_handle,
+    attribute_handle_t attribute_handle
+) {
+    return launch_procedure<ReadAttributeProcedure>(
+        connection_handle, attribute_handle
+    );
+}
+
+ble_error_t nRF5XGattClient::read_using_characteristic_uuid(
+    connection_handle_t connection_handle,
+    attribute_handle_range_t read_range,
+    const UUID& uuid
+) {
+    return launch_procedure<ReadUsingCharacteristicUUIDProcedure>(
+        connection_handle, read_range, uuid
+    );
+}
+
+ble_error_t nRF5XGattClient::read_attribute_blob(
+    connection_handle_t connection,
+    attribute_handle_t attribute_handle,
+    uint16_t offset
+) {
+    return launch_procedure<ReadAttributeBlobProcedure>(
+        connection, attribute_handle, offset
+    );
+}
+
+ble_error_t nRF5XGattClient::read_multiple_characteristic_values(
+    connection_handle_t connection,
+    const ArrayView<const attribute_handle_t>& characteristic_handles
+) {
+    return launch_procedure<ReadMultipleCharacteristicsProcedure>(
+        connection, characteristic_handles
+    );
+}
+
+ble_error_t nRF5XGattClient::write_without_response(
+    connection_handle_t connection_handle,
+    attribute_handle_t characteristic_value_handle,
+    const ArrayView<const uint8_t>& value
+) {
+    ble_gattc_write_params_t write_params = {
+        BLE_GATT_OP_WRITE_CMD,
+        /* exec flags */ 0,
+        characteristic_value_handle,
+        /* offset */ 0,
+        static_cast<uint16_t>(value.size()),
+        const_cast<uint8_t*>(value.data())
```

Comment:
what's with the juggling of consts? is const or not?

Reply:
The nordic API is not const correct; the data passed in the write params won't be modified by the API.